frontend/ts-src/signup.ts

```diff
@@ -0,0 +1,614 @@
+
+    const logo: HTMLDivElement = createDivElement(["user-logo"], `${userPrompt} >`);
+    replyContainer.appendChild(logo);
+
+    const userReply: HTMLDivElement = createDivElement(["reply"]);
+    userReply.textContent = message;
+    replyContainer.appendChild(userReply);
+
+    messageContainer.appendChild(replyContainer);
+
+    return messageContainer;
+
+}
+
+
+export function buildOptions(): HTMLDivElement {
+    /* Build HTML elements for questions with options.
+    *
+    * This is needed for multiple choice questions e.g. picking a theme.
+    *
+    * @returns { HTMLDivElement }
+    */
+
+    const options = questions[registerConfig.questionIndex].options as ThemeOptions[];
+    const optionsContainer: HTMLDivElement = createDivElement(["theme"]);
+
+    for (let i = 0; i < options.length; i++) {
+        const singleOption: HTMLDivElement = createDivElement(["theme-option"]);
+
+        const imgElement: HTMLImageElement = createImageElement([], options[i].imgUrl);
+        singleOption.appendChild(imgElement);
+
+        const nameElement: HTMLDivElement = createDivElement([], options[i].name);
+        singleOption.appendChild(nameElement);
+
+        optionsContainer.appendChild(singleOption);
+    }
+    return optionsContainer;
+}
+
+
+export function typeCharacter(
+    index: number,
+    message: string,
+    element: HTMLDivElement,
+): void {
+    /* Recursively "type" text onto the screen.
+    *
+    * @param { number } index: the index of the next character to be displayed
+    * @param { string } message: The message being displayed
+    * @param { HTMLDivElement } element: the element the message is going into
+    * @returns { void }
+    */
+
+    if (index < message.length && registerConfig.userTyping) {
+        /*
+        print everything because user is already typing. If the user
+        * presses enter, it will trigger the next question to start being
+        * displayed before the previous one is finished. It can lead to
+        * some buggy UI behaviour.
+        */
+        element.textContent += message.slice(index);
+    }
+
+    else if (index < message.length) {
+        // use this to turn settimeout into an async function and pull out options stuff:
+        // https://stackoverflow.com/questions/33289726/combination-of-async-function-await-settimeout
+        element.textContent += message.charAt(index);
+        /*
+        * This queues the function to be called later, so its
+        * "asyc-lite". This means we currently cant do the options
+        * outside of this function coz the setTimeout doesn't complete
+        * before the options block gets appended to the parent
+        * element, thus fucking everything up.
+        * I dont like this, feels like code smell. It should be fixed
+        * for now I blame the original author windowcil :D
+        */
+        setTimeout(
+            () => { typeCharacter(index + 1, message, element); },
```

Comment:
What are you trying to do? You can resolve the promise here and await it where you want to do stuff synchronously.

```
return new Promise((resolve, reject) => {
  setTimeout(() =>{
    resolve()
  }, 123)
})